src/net/server/mod.rs

```diff
@@ -55,6 +68,37 @@ impl MpcHelperServer {
     /// Starts a new instance of MPC helper and binds it to a given target.
     /// Returns a socket it is listening to and the join handle of the web server running.
     pub async fn bind(&self, target: BindTarget) -> (SocketAddr, JoinHandle<()>) {
+        async fn serve<A>(
+            server: Server<A>,
+            handle: Handle,
+            svc: IntoMakeService<Router>,
+        ) -> JoinHandle<()>
+        where
+            A: Accept<
+                    AddrStream,
+                    <IntoMakeService<Router> as MakeServiceRef<
+                        AddrStream,
+                        hyper::Request<hyper::Body>,
+                    >>::Service,
+                > + Clone
+                + Send
+                + Sync
+                + 'static,
+            A::Stream: AsyncRead + AsyncWrite + Unpin + Send,
+            A::Service: SendService<Request<hyper::Body>> + Send,
+            A::Future: Send,
```

Comment:
This arose out of an effort to reduce code duplication in the `match` statement below. Given how ugly it turned out, I'm inclined to put it back the way it was and live with the duplication of `server.handle(handle).serve(svc).await.expect("Failed to serve")`.